core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -638,6 +666,24 @@ static Parameter ofRequestObject(Class<?> type,
             this.name = name;
             this.defaultValue = defaultValue;
             this.requestConverterFunction = requestConverterFunction;
+
+            if (type.isEnum()) {
+                final Set<? extends Enum<?>> enumInstances = EnumSet.allOf(type.asSubclass(Enum.class));
+                upperCaseEnum = enumInstances.stream().collect(toImmutableMap(e -> e.name().toUpperCase(),
+                                                                              Function.identity()));
+                isEnum = true;
+            } else {
+                upperCaseEnum = null;
+                isEnum = false;
+            }
+        }
+
+        Map<String, Enum<?>> getUpperCaseEnum() {
+            return upperCaseEnum;
+        }
+
+        boolean isEnum() {
+            return isEnum;
```

Comment:
How about removing the member and just returning `type.isEnum()`?

Reply:
Checked & Updated Thanks